akka-cluster-sharding-typed/src/test/scala/akka/cluster/sharding/typed/ClusterShardingSpec.scala

```diff
@@ -150,18 +150,66 @@ class ClusterShardingSpec extends TestKit("ClusterShardingSpec", ClusterSharding
 
   val typeKey2 = EntityTypeKey[IdTestProtocol]("no-envelope-shard")
   val behaviorWithId = Behaviors.immutable[IdTestProtocol] {
-    case (_, IdStopPlz(_)) ⇒
+    case (_, IdStopPlz()) ⇒
       Behaviors.stopped
 
     case (ctx, IdWhoAreYou(_, replyTo)) ⇒
-      replyTo ! s"I'm ${ctx.self.path.name}"
+      val address = Cluster(ctx.system).selfMember.address
+      replyTo ! s"I'm ${ctx.self.path.name} at ${address.host.get}:${address.port.get}"
       Behaviors.same
 
     case (_, IdReplyPlz(_, toMe)) ⇒
       toMe ! "Hello!"
       Behaviors.same
   }
 
+  val shardingRef1: ActorRef[ShardingEnvelope[TestProtocol]] = sharding.spawn[TestProtocol](
+    _ ⇒ behavior,
+    Props.empty,
+    typeKey,
+    ClusterShardingSettings(system),
+    10,
+    StopPlz())
+
+  val shardingRef2 = sharding2.spawn[TestProtocol](
+    _ ⇒ behavior,
+    Props.empty,
+    typeKey,
+    ClusterShardingSettings(system2),
+    10,
+    StopPlz())
+
+  val shardingRef3: ActorRef[IdTestProtocol] = sharding.spawn3[IdTestProtocol, IdTestProtocol](
+    _ ⇒ behaviorWithId,
+    Props.empty,
+    typeKey2,
+    ClusterShardingSettings(system),
+    ShardingMessageExtractor.noEnvelope[IdTestProtocol](10, IdStopPlz()) {
+      case IdReplyPlz(id, _)  ⇒ id
+      case IdWhoAreYou(id, _) ⇒ id
+      case other              ⇒ throw new IllegalArgumentException(s"Unexpected message $other")
+    })
+
+  val shardingRef4 = sharding2.spawn3[IdTestProtocol, IdTestProtocol](
```

Comment:
here it's even more boilerplate since there are two type parameters, it's for the case when giving a custom extractor (no envelopes)

overloaded methods doesn't work, so we must invent different names (spawn3 is ofc not the final name)